lib/data/wrapper.js

```diff
@@ -96,31 +96,36 @@ const data = {
         // If objectGetInfo.key exists the md-model-version is 2 or greater.
         // Otherwise, the objectGetInfo is just the key string.
         const key = objectGetInfo.key ? objectGetInfo.key : objectGetInfo;
-        log.debug('sending delete to datastore', { implName, key,
+        log.debug('sending delete to dataStore', { implName, key,
             method: 'delete' });
         client.delete(key, log.getSerializedUids(), err => {
             if (err) {
-                log.error('error from sproxyd', { error: err });
+                log.error('error deleting object from dataStore',
+                    { error: err }, { location: key });
                 return cb(errors.InternalError);
             }
+            log.trace('delete object successful');
             return cb();
         });
     },
 
     // It would be preferable to have an sproxyd batch delete route to
     // replace this
-    batchDelete: (locations, log) => {
+    batchDelete: (locations, log, cb) => {
         async.eachLimit(locations, 5, (loc, next) => {
-            data.delete(loc, log, err => {
-                if (err) {
-                    log.error('one part of a batch delete failed',
-                    { location: loc });
-                }
-                return next();
-            });
+            data.delete(loc, log, () => next());
         },
-        () => {
+        err => {
+            if (err && cb) {
+                return cb(err);
+            } else if (!err) {
```

Comment:
The intention was to let data.delete() handle logging an error in case one part of the delete failed. After consulting with Rahul, I fixed my call to data.delete to use the appropriate shorthand for next(). I also added a log.error for the batchDelete and tried to clarify the code by setting a default for the callback.  I'd welcome your comments and suggestions on it. https://github.com/scality/S3/pull/324/commits/b2776fe453ffdda0e1ea4c9a9f49dd6a83402413